src/view.c

```diff
@@ -554,9 +555,21 @@ void
 view_set_decorations(struct view *view, bool decorations)
 {
 	assert(view);
-	if (view->ssd_enabled != decorations && !view->fullscreen) {
-		view->ssd_enabled = decorations;
-		ssd_update_geometry(view);
+	if (decorations) {
+		if (view->ssd) {
+			return;
+		}
+		view->ssd = ssd_create(view,
+			view == view->server->focused_view);
+	} else {
+		if (!view->ssd) {
+			return;
+		}
+		ssd_destroy(view->ssd);
+		view->ssd = NULL;
+	}
+
+	if (!view->fullscreen) {
```

Comment:
The existing logic made `view_set_decorations()` a no-op for a fullscreen view, and I am not sure why.
Also, it looks like the decorations are not actually hidden for a fullscreen view (are they just off-screen?)
So it seems like it's okay to do the `ssd_create/destroy()` part for a fullscreen view, which will result in the decorations being visually enabled/disabled once the view is no longer fullscreen. But I'd appreciate a second pair of eyes on this.

Reply:
Addressed in #650.